### PR TITLE
[yandex-maps]: adds-some-types

### PR DESCRIPTION
**Pull request by vredsun**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
**ZoomControl** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/control.ZoomControl.html/
**TypeSelector** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/control.TypeSelector.html/
**data.Manager.set** https://yandex.ru/dev/maps/archive/doc/jsapi/2.0/ref/reference/data.Manager.html#method_detail__set  in docs only `object`, but can any
**LineString.setCoordinates** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/ILineStringGeometryAccess.html/#method_detail__setCoordinates
**layout.storage** https://yandex.ru/dev/docs/search/?query=layout.storage
**map.Manager.getAll()** https://yandex.ru/dev/docs/search/?query=map.Manager.getAll
**option.presetStorage** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/option.presetStorage.html/
**Balloon** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/map.Balloon.html/?lang=ru
**ObjectManager** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/ObjectManager.html
**Hotspots** https://yandex.ru/dev/maps/jsapi/doc/2.1/ref/reference/Hotspot.html/

- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/yandex-maps/index.d.ts [hunk 1]
```diff
@@ -680,10 +735,11 @@ declare namespace ymaps {
     }
 
     namespace event {
-        class Manager implements IEventManager {
+        class Manager<TargetGeometry = {}> implements IEventManager<TargetGeometry> {
             constructor(params?: { context?: object; controllers?: IEventWorkflowController[]; parent?: IEventManager });
 
-            add(types: string[][] | string[] | string, callback: (event: (object | IEvent)) => void, context?: object, priority?: number): this;
+            add(types: 'mousedown', callback: (event: (IEvent<MouseEvent, TargetGeometry>)) => void, context?: object, priority?: number): this;
```
Comment (vredsun) [inline types/yandex-maps/index.d.ts:741]: перегрузка на `mousedown`

```typescript
pointsObjectManager.events.add('mousedown', event => { // typeof event =  IEvent<MouseEvent, {}>
```

File: types/yandex-maps/index.d.ts [hunk 2]
```diff
@@ -1574,6 +1632,8 @@ declare namespace ymaps {
                 remove(object: object): this;
 
                 getMap(): Map;
+
+                getAll(): Array<Collection<Layer>>;
```
Comment (vredsun) [inline types/yandex-maps/index.d.ts:1636]:
```typescript
  const layersCollections = map.layers.getAll(); // ymaps.Collection<ymaps.Layer>[]
  const layers = layersCollections.flatMap(collection => collection.getAll()); // ymaps.Layer[]
```

File: types/yandex-maps/index.d.ts [hunk 3]
```diff
@@ -2967,7 +3049,13 @@ declare namespace ymaps {
         behaviors?: string[];
         bounds?: number[][];
         center?: number[];
-        controls?: string[];
+        controls?: Array<
```
Comment (vredsun) [inline types/yandex-maps/index.d.ts:3052]: по факту можно `string` | ' boolean` | `control.AnyClass`

File: types/yandex-maps/index.d.ts [hunk 4]
```diff
@@ -2755,16 +2824,25 @@ declare namespace ymaps {
...
+            };
+        };
     }
 
-    class GeoObject implements IGeoObject {
+    class GeoObject<T = IGeometry, TargetGeometry = {}> implements IGeoObject<T> {
         constructor(feature?: IGeoObjectFeature, options?: IGeoObjectOptions);
 
-        geometry: IGeometry | null;
+        geometry: T | null;
```
Comment (vredsun) [inline types/yandex-maps/index.d.ts:2841]: правильный тип геометрии

**Comment by vredsun** [comment]
🔔 @Delagen @gastwork13

**Review by delagen — APPROVED** [review]

**Comment by vredsun** [comment]
Ready to merge

**Pull request merged**